src/test/java/no/unit/nva/cristin/institutions/FetchCristinInstitutionsTest.java

```diff
@@ -0,0 +1,217 @@
+        assertEquals(response.getHeaders().get(HttpHeaders.CONTENT_TYPE), MediaType.APPLICATION_JSON);
+    }
+
+    @Test
+    public void testQueryErrorResponse() throws Exception {
+        when(mockCristinApiClient.queryInstitutions(any())).thenThrow(new IOException(MOCK_EXCEPTION));
+
+        Map<String, Object> event = new HashMap<>();
+        Map<String, String> queryParams = new TreeMap<>();
+        queryParams.put(NAME_KEY, NAME_NTNU);
+        queryParams.put(LANGUAGE_KEY, LANGUAGE_NB);
+        event.put(QUERY_STRING_PARAMETERS_KEY, queryParams);
+
+        FetchCristinInstitutions mockFetchCristinInstitutions = new FetchCristinInstitutions(mockCristinApiClient);
+        GatewayResponse response = mockFetchCristinInstitutions.handleRequest(event, null);
+
+        assertEquals(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), response.getStatusCode());
+        assertEquals(response.getHeaders().get(HttpHeaders.CONTENT_TYPE), MediaType.APPLICATION_JSON);
+    }
+
+    @Test
+    public void testSuccessfulResponseWithGetInstitutionFailure() throws Exception {
+        when(mockCristinApiClient.fetchQueryInstitutionsResults(any()))
+                .thenReturn(mockQueryCristinQueryInstitutionResponseReader());
+        when(mockCristinApiClient.queryInstitutions(any())).thenCallRealMethod();
+        when(mockCristinApiClient.getInstitution(any(), any())).thenThrow(new IOException(MOCK_EXCEPTION));
+        when(mockCristinApiClient.generateQueryInstitutionsUrl(any())).thenCallRealMethod();
+
+        Map<String, Object> event = new HashMap<>();
+        Map<String, String> queryParams = new TreeMap<>();
+        queryParams.put(NAME_KEY, NAME_NTNU);
+        queryParams.put(LANGUAGE_KEY, LANGUAGE_NB);
+        event.put(QUERY_STRING_PARAMETERS_KEY, queryParams);
+
+        FetchCristinInstitutions mockFetchCristinInstitutions = new FetchCristinInstitutions(mockCristinApiClient);
+        GatewayResponse response = mockFetchCristinInstitutions.handleRequest(event, null);
+
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatusCode());
+        assertEquals(response.getHeaders().get(HttpHeaders.CONTENT_TYPE), MediaType.APPLICATION_JSON);
+    }
+
+
+    @Test
+    public void testEmptyNameParam() {
+
+        Map<String, Object> event = new HashMap<>();
+        Map<String, String> queryParams = new TreeMap<>();
+        queryParams.put(NAME_KEY, "");
+        event.put(QUERY_STRING_PARAMETERS_KEY, queryParams);
+
+        FetchCristinInstitutions mockFetchCristinInstitutions = new FetchCristinInstitutions(mockCristinApiClient);
+        GatewayResponse response = mockFetchCristinInstitutions.handleRequest(event, null);
+
+        assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), response.getStatusCode());
+        assertEquals(response.getHeaders().get(HttpHeaders.CONTENT_TYPE), MediaType.APPLICATION_JSON);
+        assertEquals(response.getBody(), "{\"error\":\"Parameter 'name' is mandatory\"}");
```

Comment:
There we have for sure an eroor text in a Constant.
Yes, here: 'FetchCristinInstitutions.NAME_IS_NULL'
and than you might use contains or even build the errorBody with help of GatewayResponses method.

```suggestion
        assertTrue(response.getBody().contains(FetchCristinInstitutions.NAME_IS_NULL);
```